File: frmCustSearch.vb
Public Class frmCustSearch
    Public vCustid As Integer = 0
    Public vLname As String = String.Empty
    Public vFname As String = String.Empty

    Private Sub frmCustSearch_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub

    Private Sub frmCustSearch_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then
            MessageBox.Show("Please Note that no customer is selected when you pressed the ESC key!")
            vCustid = 1
            vLname = "CASH"
            vFname = "CASH"
            frmPOS.cmbPaymentType.Text = "CASH"

            Me.Close()
        End If
    End Sub

    Private Sub frmCustSearch_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        txtLastname.Focus()
    End Sub

    Sub LoadCust()
        If txtLastname.Text = String.Empty Then
            MessageBox.Show("Enter a lastname to seaarch !")
            txtLastname.Focus()
            Exit Sub
        Else

            Dim mgrmembers As membersManager
            Dim cols As membersCollection
            Try
                mgrmembers = New membersManager
                mgrmembers.DataObject.SelectFilter = membersData.SelectFilters.ListBox
                mgrmembers.DataObject.WhereFilter = membersData.WhereFilters.Likelastname
                mgrmembers.Entity.lastname = txtLastname.Text.ToString
                mgrmembers.DataObject.OrderByFilter = membersData.OrderByFilters.lastname
                cols = mgrmembers.BuildCollection()
                If mgrmembers.DataObject.RowsAffected > 0 Then
                    dgcustomer.DataSource = cols
                    dgcustomer.Columns(0).Width = 0
                    dgcustomer.Columns(1).Width = 300
                    dgcustomer.Columns(2).Width = 300
                    dgcustomer.Columns(3).Width = 50

                    dgcustomer.Focus()
                Else
                    MessageBox.Show("No customers related to the search criteria!")
                End If

            Catch ex As PDSAValidationException
                '    MessageBox.Show(ex.Message)
            Catch ex As Exception
                MessageBox.Show(ex.ToString())
            End Try
        End If
    End Sub

    Private Sub btnSearch_Click(sender As Object, e As EventArgs) Handles btnSearch.Click
        LoadCust()
    End Sub

    Private Sub txtLastname_KeyDown(sender As Object, e As KeyEventArgs) Handles txtLastname.KeyDown
        If e.KeyCode = Keys.Down Then
            dgcustomer.Focus()
        End If
        If e.KeyCode = Keys.Right Then
            btnSearch.Focus()
        End If
    End Sub

    Private Sub txtLastname_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtLastname.KeyPress
        If e.KeyChar = Chr(13) Then
            LoadCust()
        End If
    End Sub

    Private Sub btnClose_Click(sender As Object, e As EventArgs) Handles btnClose.Click
        'Me.Close()
    End Sub

    Private Sub dgcustomer_KeyDown(sender As Object, e As KeyEventArgs) Handles dgcustomer.KeyDown
        If e.KeyCode = Keys.Enter Then

            If dgcustomer.Rows.Count < 1 Then
                Exit Sub
            End If
            Try
                Dim vRow As Integer = 0
                Dim vint As Integer = 0
                Dim x1 As String = String.Empty
                Dim x2 As String = String.Empty
                vRow = dgcustomer.CurrentRow.Index
                vint = CInt(dgcustomer(0, vRow).Value)
                x1 = CStr(dgcustomer(1, vRow).Value)
                x2 = CStr(dgcustomer(2, vRow).Value)
                vCustid = vint
                vLname = x1
                vFname = x2

                Me.Close()
            Catch ex As Exception
                MessageBox.Show(ex.ToString)
            End Try
            e.SuppressKeyPress = True
        End If

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        txtLastname.Focus()
    End Sub
End Class

File: DoorsPOS.WinApp/CS/frmCustSearch.cs
// VBConversions Note: VB project level imports
using System.Data;
using System.Collections.Generic;
using PDSA.Data;
using PDSA.WinForms.Forms;
using PDSA.Security;
using System.Diagnostics;
using PDSA.Common;
using PDSA.WinForms;
using PDSA.WinForms.Security;
using PDSA.PDSACryptography;
using DoorsPOS.Utility;
using Microsoft.VisualBasic;
using System.Collections;
using System;
using DoorsPOS.BusinessLayer;
using System.Windows.Forms;
using System.Drawing;
using System.Linq;
using DoorsPOS.DataLayer;
// End of VB project level imports


namespace DoorsPOS.WinApp
{
	public partial class frmCustSearch
	{
		public frmCustSearch()
		{
			InitializeComponent();
		}
		public int vCustid = 0;
		public string vLname = string.Empty;
		public string vFname = string.Empty;
		public void frmCustSearch_FormClosed(object sender, FormClosedEventArgs e)
		{
			this.Dispose();
		}
		
		public void frmCustSearch_KeyDown(object sender, KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Escape)
			{
				MessageBox.Show("Please Note that no customer is selected when you pressed the ESC key!");
				vCustid = 205;
				vLname = "CASH";
				vFname = "CASH";
				this.Close();
			}
		}
		
		public void frmCustSearch_Load(object sender, EventArgs e)
		{
			txtLastname.Focus();
		}
		
		public void LoadCust()
		{
			if (txtLastname.Text == string.Empty)
			{
				MessageBox.Show("Enter a lastname to seaarch !");
				txtLastname.Focus();
				return;
			}
			else
			{
				
				membersManager mgrmembers = default(membersManager);
				membersCollection cols = default(membersCollection);
				try
				{
					mgrmembers = new membersManager();
					mgrmembers.DataObject.SelectFilter = membersData.SelectFilters.ListBox;
					mgrmembers.DataObject.WhereFilter = membersData.WhereFilters.Likelastname;
					mgrmembers.Entity.lastname = txtLastname.Text.ToString();
					mgrmembers.DataObject.OrderByFilter = membersData.OrderByFilters.lastname;
					cols = mgrmembers.BuildCollection();
					if (mgrmembers.DataObject.RowsAffected > 0)
					{
						dgcustomer.DataSource = cols;
						dgcustomer.Columns[0].Width = 0;
						dgcustomer.Columns[1].Width = 300;
						dgcustomer.Columns[2].Width = 300;
						dgcustomer.Columns[3].Width = 50;
						dgcustomer.Focus();
					}
					else
					{
						MessageBox.Show("No customers related to the search criteria!");
					}
					
				}
				catch (PDSAValidationException)
				{
					//    MessageBox.Show(ex.Message)
				}
				catch (Exception ex)
				{
					MessageBox.Show(ex.ToString());
				}
			}
		}
		
		public void btnSearch_Click(object sender, EventArgs e)
		{
			LoadCust();
		}
		
		public void txtLastname_KeyDown(object sender, KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Down)
			{
				dgcustomer.Focus();
			}
			if (e.KeyCode == Keys.Right)
			{
				btnSearch.Focus();
			}
		}
		
		public void txtLastname_KeyPress(object sender, KeyPressEventArgs e)
		{
			if (e.KeyChar == '\r')
			{
				LoadCust();
			}
		}
		
		public void btnClose_Click(object sender, EventArgs e)
		{
			this.Close();
		}
		
		public void dgcustomer_KeyDown(object sender, KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Enter)
			{
				
				if (dgcustomer.Rows.Count < 1)
				{
					return;
				}
				try
				{
					int vRow = 0;
					int vint = 0;
					string x1 = string.Empty;
					string x2 = string.Empty;
					vRow = dgcustomer.CurrentRow.Index;
					vint = System.Convert.ToInt32(dgcustomer[0, vRow].Value);
					x1 = (dgcustomer[1, vRow].Value).ToString();
					x2 = (dgcustomer[2, vRow].Value).ToString();
					vCustid = vint;
					vLname = x1;
					vFname = x2;
					this.Close();
				}
				catch (Exception ex)
				{
					MessageBox.Show(ex.ToString());
				}
				e.SuppressKeyPress = true;
			}
			
		}
		
		public void Button1_Click(object sender, EventArgs e)
		{
			txtLastname.Focus();
		}
	}
}
